### PR TITLE
bug - Nokia discovery protocols

### LINKED ISSUE #15598: Error discovering discovery-protocols module (Nokia Timos)
### The problem

According to device log inside LibreNMS Gui i'm having this issue on every timos device since just after midnight on12th of november.
I do see "neighbours"  information on every device in gui so i'm not sure if this is correct or not.

### Output of ./validate.php

```txt
===========================================
Component | Version
--------- | -------
LibreNMS  | 23.11.0-6-ga61c11db7 (2023-11-23T02:53:19+01:00)
DB Schema | 2023_11_21_172239_increase_vminfo.vmwvmguestos_column_length (274)
PHP       | 8.1.11
Python    | 3.8.10
Database  | MariaDB 10.3.36-MariaDB-1:10.3.36+maria~ubu1804
RRDTool   | 1.7.2
SNMP      | 5.8
===========================================

[OK]    Composer Version: 2.6.5
[OK]    Dependencies up-to-date.
[OK]    Database connection successful
[OK]    Database Schema is current
[OK]    SQL Server meets minimum requirements
[OK]    lower_case_table_names is enabled
[OK]    MySQL engine is optimal
[OK]    Database and column collations are correct
[OK]    Database schema correct
[OK]    MySQl and PHP time match
[OK]    Active pollers found
[OK]    Dispatcher Service not detected
[OK]    Locks are functional
[OK]    Python poller wrapper is polling
[WARN]  Using database for locking, you should set CACHE_DRIVER=redis
[OK]    rrd_dir is writable
[OK]    rrdtool version ok
[WARN]  Your local git contains modified files, this could prevent automatic updates.
        [FIX]: 
        You can fix this with ./scripts/github-remove
        Modified Files:
         rrd/.gitignore
```

### What was the last working version of LibreNMS?

_No response_

### Anything in the logs that might be useful for us?

```txt
Error discovering discovery-protocols module for 10.10.12.211. TypeError: key(): Argument #1 ($array) must be of type array, null given in /home/librenms/includes/discovery/discovery-protocols.inc.php:170
key(): Argument #1 ($array) must be of type array, null given {"exception":"[object] (TypeError(code: 0): key(): Argument #1 ($array) must be of type array, null given at /home/librenms/includes/discovery/discovery-protocols.inc.php:170)"}
```

### PR DESCRIPTION
**Pull request by PipoCanaja**
Trying to fix the nokia discovery issue

fixes #15598 

DO NOT DELETE THE UNDERLYING TEXT

#### Please note

> Please read this information carefully. You can run `./lnms dev:check` to check your code before submitting.

- [x] Have you followed our [code guidelines?](https://docs.librenms.org/Developing/Code-Guidelines/)
- [x] If my Pull Request does some changes/fixes/enhancements in the WebUI, I have inserted a screenshot of it.
- [ ] If my Pull Request makes discovery/polling/yaml changes, I have added/updated [test data](https://docs.librenms.org/Developing/os/Test-Units/).

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`
After you are done testing, you can remove the changes with `./scripts/github-remove`.  If there are schema changes, you can ask on discord how to revert.

### CODE REVIEW

**Comment by PipoCanaja** [comment]
Hi @dejantep 
Could you have a look at this PR and see if it fixes the crash ? I don't have any nokia so I can't fix any discovery issue. Just trying to fix the exception here.
Bye

**Comment by dejantep** [comment]
Hello @PipoCanaja 
I have applied it and it seems to work BUT now tab "neighbours" is missing completely from device view.

**Comment by PipoCanaja** [comment]
Ok so that means that this variable always comes empty. Are you sure the device is providing the neighbours in the LLDP table ? Could you check with an snmpwalk  ?

**Comment by dejantep** [comment]
yes it does. it was working before. here's snippet of snmpwalk

```
iso.3.6.1.4.1.6527.3.1.2.59.4.1.1.8.22564.35700737.1.1 = STRING: "1/1/c2/1, 100-Gig Ethernet"
iso.3.6.1.4.1.6527.3.1.2.59.4.1.1.8.22570.37453824.1.2 = STRING: "1/1/25, 10-Gig Ethernet"
iso.3.6.1.4.1.6527.3.1.2.59.4.1.1.8.22576.35733505.1.3 = STRING: "1/1/c2/1, 100-Gig Ethernet"
iso.3.6.1.4.1.6527.3.1.2.59.4.1.1.8.1068754457.37060608.1.4 = STRING: "2/1/11, 10-Gig Ethernet, \"Core:to-mr14\""
iso.3.6.1.4.1.6527.3.1.2.59.4.1.1.8.2401180442.35799041.1.6 = STRING: "1/1/c1/1, 100-Gig Ethernet"
iso.3.6.1.4.1.6527.3.1.2.59.4.1.1.9.22564.35700737.1.1 = STRING: "mr14-core-rtr-01"
iso.3.6.1.4.1.6527.3.1.2.59.4.1.1.9.22570.37453824.1.2 = STRING: "7210-dist"
iso.3.6.1.4.1.6527.3.1.2.59.4.1.1.9.22576.35733505.1.3 = STRING: "mr14-core-pe-01"
iso.3.6.1.4.1.6527.3.1.2.59.4.1.1.9.1068754457.37060608.1.4 = STRING: "mr25-dist-pe-01"
iso.3.6.1.4.1.6527.3.1.2.59.4.1.1.9.2401180442.35799041.1.6 = STRING: "tdn1-core-pe-01

```

**Comment by PipoCanaja** [comment]
Could be that part of the data is missing (and triggering the NULL exception). 
Would you be OK to provide me an snmprec file of your device. You can send it to me directly if you don't want to publish it here. I will then be able to reproduce exactly your situation. 
You can do it with `./scripts/collect-snmp-data.php  -h <deviceID> -v pipocanaja` and `snmpwalk -v2c -OUneb -c <community> <ipOfDevice> . > capture.ouneb.txt` and share both resulting files with me. (be aware that these files will probably disclose details of your network, so better to use LAB devices). 1st file may be enough as a 1st step, 2nd file is a complete dump so best is to provide both if you can.

**Comment by dejantep** [comment]
I have a lab device that can be used, so yes.
Shall i remove applied PR so i can go back till "normal"?

**Comment by PipoCanaja** [comment]
> I have a lab device that can be used, so yes. Shall i remove applied PR so i can go back to "normal"?
Should not have any impact so you can stay as it is

> hmm..i'm getting error regarding collect-snmp-data stating this `LibreNMS\Util\ModuleTestHelper::captureFromDevice(): Argument #1 ($device_id) must be of type int, null given` This device is not in LibreNMS
collect-snmp-data expects a device_id, so yes, the device must be in LibreNMS (added, discovered).

> i have 2nd file ready. how can i send it to you?
You can contact me on the LibreNMS Discord, or on https://community.librenms.org/

File: includes/discovery/discovery-protocols.inc.php [hunk 1]
```diff
@@ -140,60 +140,64 @@
...
-        $remote_device_id = find_device_id($lldp['tmnxLldpRemSysName'][$ifIndex][$MacIndex][$RemIndex]);
-
-        if (! $remote_device_id &&
-            \LibreNMS\Util\Validate::hostname($lldp['tmnxLldpRemSysName'][$ifIndex][$MacIndex][$RemIndex]) &&
-            ! can_skip_discovery($lldp['tmnxLldpRemSysName'][$ifIndex][$MacIndex][$RemIndex], $lldp['tmnxLldpRemSysDesc'][$ifIndex][$MacIndex][$RemIndex]) &&
-            Config::get('autodiscovery.xdp') === true
-        ) {
-            $remote_device_id = discover_new_device($lldp['tmnxLldpRemSysName'][$ifIndex][$MacIndex][$RemIndex], $device, 'LLDP', $interface);
-        }
+    $lldp_array = SnmpQuery::hideMib()->walk('TIMETRA-LLDP-MIB::tmnxLldpRemoteSystemsData')->table(4);
```
Comment (PipoCanaja) [inline includes/discovery/discovery-protocols.inc.php:168]: Use of table(4) to get a correctly structured associative array. Makes everything simpler afterward.

**Comment by dejantep** [comment]
@PipoCanaja do i need to do anything else about this so it gets merged?

**Comment by PipoCanaja** [comment]
> @PipoCanaja do i need to do anything else about this so it gets merged?

Not on your side. LibreNMS team of maintainers will review it and hopefully merge it :)

**Comment by craig-nokia** [comment]
Fix is verified.

**Pull request merged**